python/cudf/cudf/tests/test_concat.py

```diff
@@ -593,3 +593,421 @@ def test_concat_dataframe_with_multiIndex(df1, df2):
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+        ),
+        gd.concat(
+            [gdf1, gdf2, gdf3, gdf_empty1],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+        ),
+    )
+
+
+@pytest.mark.parametrize("ignore_index", [True, False])
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("join", ["inner", "outer"])
+def test_concat_join_one_df(ignore_index, sort, join):
+    pdf1 = pd.DataFrame(
+        {
+            "x": range(10),
+            "y": list(map(float, range(10))),
+            "z": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        }
+    )
+
+    gdf1 = gd.from_pandas(pdf1)
+
+    assert_eq(
+        pd.concat([pdf1], sort=sort, join=join, ignore_index=ignore_index),
+        gd.concat([gdf1], sort=sort, join=join, ignore_index=ignore_index),
+    )
+
+
+@pytest.mark.parametrize("ignore_index", [True, False])
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("join", ["inner", "outer"])
+@pytest.mark.parametrize("axis", [1])
+def test_concat_join_no_overlapping_columns(ignore_index, sort, join, axis):
+    pdf4 = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    pdf5 = pd.DataFrame({"c": [7, 8, 9], "d": [10, 11, 12]})
+
+    gdf4 = gd.from_pandas(pdf4)
+    gdf5 = gd.from_pandas(pdf5)
+
+    assert_eq(
+        pd.concat(
+            [pdf4, pdf5],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ),
+        gd.concat(
+            [gdf4, gdf5],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ),
+    )
+
+
+@pytest.mark.parametrize("ignore_index", [False, True])
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("join", ["inner", "outer"])
+@pytest.mark.parametrize("axis", [0])
+# leaving out axis = 1 because pandas returns object index not range index
+def test_concat_join_no_overlapping_columns_many_and_empty_axis_0(
+    ignore_index, sort, join, axis
+):
+    pdf4 = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    pdf5 = pd.DataFrame({"c": [7, 8, 9], "d": [10, 11, 12]})
+    pdf6 = pd.DataFrame(
+        {
+            "x": range(10),
+            "y": list(map(float, range(10))),
+            "z": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        }
+    )
+    pdf_empty = pd.DataFrame()
+
+    gdf4 = gd.from_pandas(pdf4)
+    gdf5 = gd.from_pandas(pdf5)
+    gdf6 = gd.from_pandas(pdf6)
+    gdf_empty = gd.from_pandas(pdf_empty)
+
+    assert_eq(
+        pd.concat(
+            [pdf4, pdf5, pdf6, pdf_empty],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ),
+        gd.concat(
+            [gdf4, gdf5, gdf6, gdf_empty],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ),
+    )
```

Comment:
```suggestion
@pytest.mark.parametrize("ignore_index", [False, True])
@pytest.mark.parametrize("sort", [True, False])
@pytest.mark.parametrize("join", ["inner", "outer"])
@pytest.mark.parametrize("axis", [0, 1])
def test_concat_join_no_overlapping_columns_many_and_empty_axis_0(
    ignore_index, sort, join, axis
):
    pdf4 = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
    pdf5 = pd.DataFrame({"c": [7, 8, 9], "d": [10, 11, 12]})
    pdf6 = pd.DataFrame(
        {
            "x": range(10),
            "y": list(map(float, range(10))),
            "z": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
        }
    )
    pdf_empty = pd.DataFrame()

    gdf4 = gd.from_pandas(pdf4)
    gdf5 = gd.from_pandas(pdf5)
    gdf6 = gd.from_pandas(pdf6)
    gdf_empty = gd.from_pandas(pdf_empty)

    expected = pd.concat(
        [pdf4, pdf5, pdf6, pdf_empty],
        sort=sort,
        join=join,
        ignore_index=ignore_index,
        axis=axis,
    )
    actual = gd.concat(
        [gdf4, gdf5, gdf6, gdf_empty],
        sort=sort,
        join=join,
        ignore_index=ignore_index,
        axis=axis,
    )
    assert_eq(
        expected,
        actual,
        check_index_type=False if len(expected) == 0 else True,
    )
```

We can set `check_index_type` when we have an empty index to avoid asserting when pandas returns empty index of type `object` vs empty RangeIndex like we return.

This helps surface an issue related to index mismatch too but in `outer` join.

Reply:
Incorporated this change. Once we do this I can remove some of the other test cases that I'd separated because of this.